src/test_application/main.cpp

```diff
@@ -199,8 +200,34 @@ void load_a_bunch_of_stuff()
     osp::Package lazyDebugPack("lzdb", "lazy-debug");
 
     // Load sturdy glTF files
-    osp::AssetImporter::load_sturdy_file("OSPData/adera/spamcan.sturdy.gltf", lazyDebugPack);
-    osp::AssetImporter::load_sturdy_file("OSPData/adera/stomper.sturdy.gltf", lazyDebugPack);
+    const std::string_view datapath = {"OSPData/adera/"};
+    const std::vector<std::string_view> meshes = 
```

Comment:
This is a good example of the confusing way string lifetime works.

This is, of course, fine, since the std::string_view will always point to a string that is alive, since these strings are in the read-only data section

but if you pass these strings to a function that want to take ownership of the string data's lifetime, then it'll incur an allocate and copy.

This is why I made that shared_string class, because it provides explicit ownership behavior that also handles read-only-data (by having a default constructed m_lifetime). That allows for lifetime managing functions to accept the string whether it's in RO-Data or not, and get the optimal outcome in terms of runtime cost.

Reply:
I don't understand; is there anything wrong with what's going on here? The strings here only exist to be catted together and passed to the `load_sturdy_file()` function which is going to end up needing an allocate anyways, and whenever a function being called here requires ownership of the string argument, it just converts it into a new `std::string`. Is there something I should change to make this better? I guess you're just saying that if one of these strings were to be copied into multiple places, it could benefit from sharing the underlying memory?

Reply:
Nothing you've got here is wrong, i was just using it as an example of where string ownership can be tricky.